[Logs] Added a new page for Log Filters. (#4393)
cloudflare/cloudflare-docs: content/logs/reference/logpush-api-configuration/filters.md

---
pcx-content-type: how-to
title: Filters
weight: 40
---

# Filters

The following table represents the comparison operators that are supported and example values. Filters are added as escaped JSON strings formatted as `{\"key\":\"<field>\",\"operator\":\"<comparison_operator>\",\"value\":\"<value>\"}`.

- Fields specify properties associated with an event.

- Comparison operators define how values must relate to actual request data for an expression to return true.

- Values represent the data associated with fields. When evaluating a rule, Cloudflare compares these values with the actual data obtained from the request.

Filters can be connected using `AND`, `OR` logical operators.

> {{<table-wrap style="width:100%">}}

Reviewer: The width of this table is weird. Maybe add overview to page type and then wrap all text (minus table) in `{{<content-column>}}`?

      </td>
    </tr>
    <tr>
      <td>Value is in<br />a set of values</td>
      <td><code class="InlineCode">in</code></td>
      <td><code class="InlineCode"></code></td>
      <td>&#x2705;</td>
      <td>&#x2705;</td>
      <td>&#10060;</td>
      <td>&#x2705;</td>
      <td>&#10060;</td>
      <td>
         <code class="InlineCode">{\"key\":\"EdgeResponseStatus\",\"operator\":\"<strong>in</strong>\",\"value\":[200,201]}</code>
      </td>
    </tr>
    <tr>
      <td>Value is not<br />in a set of values</td>
      <td><code class="InlineCode">!in</code></td>
      <td><code class="InlineCode"></code></td>
      <td>&#x2705;</td>
      <td>&#x2705;</td>
      <td>&#10060;</td>
      <td>&#x2705;</td>
      <td>&#10060;</td>
      <td>
         <code class="InlineCode">{\"key\":\"EdgeResponseStatus\",\"operator\":\"<strong>!in</strong>\",\"value\":[200,201]}</code>
      </td>
    </tr>
  </tbody>
</table>
{{</table-wrap>}}

The filter field has limits of approximately 30 operators and 1000 bytes. Anything exceeding this value will return an error.

{{<Aside type="note" header="Note">}}

> We do not support filtering on the following data types: objects, array[int], array[object].

Reviewer: ```suggestion
Cloudflare does not support filtering on the following data types: objects, array[int], array[object].
```

Reviewer: Not sure if it should be Cloudflare or Logs. Also, do we need to place the data types in between backticks?

{{</Aside>}}

> For the Firewall events dataset, we do not support filtering on the following fields: Kind, MatchIndex, Metadata, OriginatorRayID, RuleID, Source.

Reviewer: ```suggestion
For the Firewall events dataset, Cloudflare does not support filtering on the following fields: Kind, MatchIndex, Metadata, OriginatorRayID, RuleID, Source.
```

> For the Gateway HTTP dataset, we do not support filtering on the following fields: Downloaded File Names, Uploaded File Names.

Reviewer: ```suggestion
For the Gateway HTTP dataset, Cloudflare does not support filtering on the following fields: Downloaded File Names, Uploaded File Names.
```

---
pcx-content-type: how-to
type: overview
title: Filters
weight: 40
layout: list
---

# Filters

> The following table represents the comparison operators that are supported and example values. Filters are added as escaped JSON strings formatted as `{\"key\":\"<field>\",\"operator\":\"<comparison_operator>\",\"value\":\"<value>\"}`.

Reviewer 2: can we replace the sample format here to {"key":"<field>","operator":"<comparison_operator>","value":"<value>"} instead

      </td>
    </tr>
    <tr>
      <td>Value is not<br />in a set of values</td>
      <td><code class="InlineCode">!in</code></td>
      <td><code class="InlineCode"></code></td>
      <td>&#x2705;</td>
      <td>&#x2705;</td>
      <td>&#10060;</td>
      <td>&#x2705;</td>
      <td>&#10060;</td>
      <td>
         <code class="InlineCode">{\"key\":\"EdgeResponseStatus\",\"operator\":\"<strong>!in</strong>\",\"value\":[200,201]}</code>
      </td>
    </tr>
  </tbody>
</table>
{{</table-wrap>}}

The filter field has limits of approximately 30 operators and 1000 bytes. Anything exceeding this value will return an error.

{{<Aside type="note" header="Note">}}
Filtering is not supported on the following data types: `objects`, `array[int]`, `array[object]`.
{{</Aside>}}

For the Firewall events dataset, the following fields are not supported: Kind, MatchIndex, Metadata, OriginatorRayID, RuleID, Source.

For the Gateway HTTP dataset, the following fields are not supported: Downloaded File Names, Uploaded File Names.

Example request using cURL:

```bash
curl -s -X POST https://api.cloudflare.com/client/v4/zones/<ZONE_ID>/logpush/jobs/ \

> -H 'X-Auth-Key: <KEY> \

Reviewer 2: missing an end quote here

-H 'X-Auth-Email:<EMAIL>' \
-H 'Content-Type: application/json' \
-d '{

> "name":"Burritobot static assets"

Reviewer 2: missing a comma at the end of this line
